planner/core/rule_count_star_rewriter.go

```diff
@@ -47,15 +47,16 @@ func (c *countStarRewriter) optimize(ctx context.Context, p LogicalPlan, opt *lo
 func (c *countStarRewriter) countStarRewriter(p LogicalPlan, opt *logicalOptimizeOp) (LogicalPlan, error) {
 	// match pattern agg(count(*)) -> datasource
 	if agg, ok := p.(*LogicalAggregation); ok {
-		// todo if all kinds of data source could be matched ?
-		if dataSource, ok := agg.Children()[0].(*DataSource); ok {
-			for _, aggFunc := range agg.AggFuncs {
-				if aggFunc.Name == "count" && len(aggFunc.Args) == 1 {
-					if constExpr, ok := aggFunc.Args[0].(*expression.Constant); ok {
-						if constExpr.Value.GetInt64() == 1 {
-							if len(dataSource.Columns) > 0 {
-								rewriteCount1ToCountColumn(dataSource, aggFunc)
-								continue
+		if agg.GroupByItems == nil || len(agg.GroupByItems) == 0 {
```

Comment:
I think we can remove `agg.GroupByItems == nil`. because len(nil) is 0, right?

Reply:
thanks for reminding